libs/core/form/form-control/form-control.component.ts

```diff
@@ -1,85 +1,176 @@
 import {
-    Attribute,
     ChangeDetectionStrategy,
     Component,
+    computed,
+    effect,
     ElementRef,
-    HostBinding,
+    inject,
+    input,
     Input,
     OnChanges,
     OnDestroy,
     OnInit,
+    Renderer2,
+    signal,
     ViewEncapsulation
 } from '@angular/core';
 import { FormStates } from '@fundamental-ngx/cdk/forms';
-import { CssClassBuilder, Nullable, applyCssClass } from '@fundamental-ngx/cdk/utils';
+import { applyCssClass, CssClassBuilder, Nullable } from '@fundamental-ngx/cdk/utils';
 import { ContentDensityObserver, contentDensityObserverProviders } from '@fundamental-ngx/core/content-density';
+import { ValueStateAriaMessageService } from '@fundamental-ngx/core/shared';
 import { Subscription } from 'rxjs';
 import { FormItemControl, registerFormItemControl } from '../form-item-control/form-item-control';
 
-/**
- * Directive intended for use on form controls.
- *
- * ```html
- * <input type="text" fd-form-control />
- * ```
- */
+let formControlId = 0;
+
 @Component({
     // eslint-disable-next-line @angular-eslint/component-selector
     selector: 'input[fd-form-control], textarea[fd-form-control]',
-    template: ` <ng-content></ng-content>`,
+    template: `<ng-content></ng-content>`,
     styleUrl: './form-control.component.scss',
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [registerFormItemControl(FormControlComponent), contentDensityObserverProviders()]
+    providers: [registerFormItemControl(FormControlComponent), contentDensityObserverProviders()],
+    host: {
+        '[attr.type]': 'type()',
+        '[attr.aria-label]': 'ariaLabel',
+        '[attr.aria-labelledby]': 'ariaLabelledBy',
+        '[attr.aria-describedby]': 'combinedAriaDescribedBy()'
+    }
 })
 export class FormControlComponent implements CssClassBuilder, OnInit, OnChanges, OnDestroy, FormItemControl {
+    /** aria-label for form-control. */
+    @Input()
+    ariaLabelledBy: Nullable<string>;
+
+    /** aria-label for form-control. */
+    @Input()
+    ariaLabel: Nullable<string>;
+
     /**
      *  The state of the form control - applies css classes.
      *  Can be `success`, `error`, `warning`, `information` or blank for default.
      */
-    @Input()
-    state: FormStates | null = null;
+    state = input<FormStates | null>(null);
 
     /** Type of the form control. */
-    @HostBinding('attr.type')
-    @Input()
-    type: string;
+    type = input<string>();
 
-    /** user's custom classes */
-    @Input()
-    class: string;
+    /** User's custom classes */
+    class = input<string>();
 
-    /** aria-label for form-control. */
-    @Input()
-    ariaLabel: Nullable<string>;
+    /** Default ARIA message text for the "success" value state. */
+    valueStateSuccessMessage = input<string>(inject(ValueStateAriaMessageService).success);
```

Comment:
Not the responsibility of this PR, but I'm going to open an issue to move away from this weird way of getting these success/info/warning/error messages. I'm guessing these aren't even translated